src/loader-new.js

```diff
@@ -0,0 +1,342 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from './log';
+import {htmlFor, htmlRefs, svgFor} from './static-template';
+import {isExperimentOn} from './experiments';
+import {toWin} from './types';
+
+/**
+ * Creates a default "loading indicator" element based on the new design.
+ *
+ * Please see https://github.com/ampproject/amphtml/issues/20237 for details,
+ * screenshots and various states of the new loader design.
+ *
+ * @param {!Document} doc
+ * @param {!Element} loadingContainer
+ * @param {!Element} element
+ * @return {!Element} new loader root element
+ */
+export function createLoaderElement(doc, loadingContainer, element) {
+  dev().assert(!isLoaderIneligible(element));
+  dev().assert(isNewLoaderExperimentEnabled(toWin(doc.defaultView)));
+
+  const loader = new LoaderBuilder(doc, loadingContainer, element);
+  return loader.build();
+}
+
+/**
+ * Helper class to build the new loader's DOM.
+ */
+class LoaderBuilder {
+  /**
+   * @param {!Document} doc
+   * @param {!Element} loadingContainer
+   * @param {!Element} element
+   */
+  constructor(doc, loadingContainer, element) {
+    /** @private {!Document} */
+    this.doc_ = doc;
+
+    /** @private {!Element} */
+    this.loadingContainer_ = loadingContainer;
+
+    /** @private {!Element} */
+    this.element_ = element;
+
+    /** @private {?Element} */
+    this.domRoot_;
+
+    /** @private {?Element} */
+    this.svgRoot_;
+  }
+
+  /**
+   * Builds the loader's DOM and returns the element.
+   * @return {!Element} new loader root element
+   */
+  build() {
+    this.buildContainers_();
+    this.setSize_();
+    this.addSpinner_();
+    this.maybeAddLogo_();
+    this.maybeAddBackgroundShim_();
+    this.maybeAddDefaultPlaceholder_();
+    return dev().assertElement(this.domRoot_);
+  }
+
+  /**
+   * Builds the wrappers for the loader.
+   * @private
+   */
+  buildContainers_() {
+    this.domRoot_ = htmlFor(this.element_)`<div class="i-amphtml-new-loader">
+      <div ref="innerContainer"></div>
+    </div>`;
+
+    /**
+     * There is an extra inner div here for backward compatibility with
+     * customizing loaders. The common and documented CSS for customizing
+     * loaders includes a style to hide the old three dots via:
+     *  .my-custom-loader .amp-active > div {
+     *     display: none;
+     *  }
+     * The extra div mimic a similar DOM.
+     */
+    const innerContainer = dev().assertElement(
+      htmlRefs(this.domRoot_)['innerContainer']
+    );
+
+    this.svgRoot_ = svgFor(
+      this.element_
+    )`<svg xmlns="http://www.w3.org/2000/svg" viewBox="24 24 72 72"></svg>`;
```

Comment:
No need for this `svgFor`, you could just put this inside the `domRoot_`'s `innerContainer` (and use a ref, if you want to reference it).